api/types/app.go

```diff
@@ -486,6 +512,28 @@ func (a *AppV3) checkTCPPorts() error {
 	return nil
 }
 
+func (a *AppV3) checkMCP() error {
+	switch GetMCPServerTransportType(a.Spec.URI) {
+	case MCPTransportStdio:
+		return trace.Wrap(a.checkMCPStdio())
+	default:
+		return trace.BadParameter("unsupported MCP server %q with URI %q", a.GetName(), a.Spec.URI)
+	}
+}
+
+func (a *AppV3) checkMCPStdio() error {
+	if a.Spec.MCP == nil {
+		return trace.BadParameter("MCP server %q missing mcp spec", a.GetName())
+	}
+	if a.Spec.MCP.Command == "" {
+		return trace.BadParameter("MCP server %q missing command", a.GetName())
+	}
+	if a.Spec.MCP.RunAsLocalUser == "" {
+		return trace.BadParameter("MCP server %q missing run_as_local_user", a.GetName())
```

Comment:
The meaning of `run_as_local_user` isn't as obvious as the other params, at least to myself. Maybe something longer, for example:

```suggestion
		return trace.BadParameter("MCP server %q must specify username to use through the run_as_local_user field", a.GetName())
```